cpp/src/spatial/hausdorff.cu

```diff
@@ -80,18 +85,23 @@ struct hausdorff_functor {
                                            result_view.begin<T>(),
                                            stream);
 
-    return result;
+    thrust::host_vector<cudf::size_type> splits(num_spaces - 1);
+    thrust::sequence(thrust::host, splits.begin(), splits.end(), num_spaces, num_spaces);
+
+    return {std::move(result),
+            cudf::table_view(cudf::detail::split(result->view(), splits, stream))};
```

Comment:
Shouldn't be calling into cudf::detail...

Reply:
Only detail API exposes the stream parameter though...

Reply:
The only place where cuDF uses stream is with the null counting on device. Here we don't need to count nulls since the result doesn't contain any.